src/transformers/models/gemma2/modeling_gemma2.py

```diff
@@ -338,7 +339,8 @@ def forward(
                 "sliding_window": self.sliding_window,
                 "cache_position": cache_position,
             }
-            key_states, value_states = past_key_value.update(key_states, value_states, self.layer_idx, cache_kwargs)
```

Comment:
See here how we pass the `cache_kwargs` to the update function of the hybrid cache, where `cache_kwargs` are defined as `{..., "sliding_window": self.sliding_window"}` a few lines above. Hence, there should be no need to pass `sliding_window` explicitly!

Reply:
#31775 
There is a similar problem to mine here. I don't think it's an issue with my personal device, as I've debugged it multiple times and this phenomenon still occurs.
"The cache was created with alternating max seq lenghts of 4k and 8k, but all layers were being updated as if they were 8k, causing out-of-bounds errors and CUDA exceptions."

Reply:
Problem is with the function definition:

```python
    def update(
        self,
        key_states: torch.Tensor,
        value_states: torch.Tensor,
        layer_idx: int,
        cache_kwargs: Optional[Dict[str, Any]] = None,
        sliding_window: Optional[int] = None,
    )
```

`cache_kwargs` is not actually a kwargs argument (missing the `**`), it's just a dict. Adding a `sliding_window` key to it doesn't affect the separate `sliding_window` argument to the update function, which just keeps its default value.

Reply:
yeah that's a good catch, none of the cache kwargs have been kwargs up until this point. and we never did `**cache_kwargs`.  I'll update this in another PR if you don't have time to fix. TBH we should remove sliding window from the update kwargs !

Reply:
Thank you, is there a problem with the order of alternating implementation of the SWA layer?

Reply:
Yeah, it's flipped. SWA should be on the even layers according to [gemma_pytorch](https://github.com/google/gemma_pytorch/blob/1814f8d0a6ba93b875c46a64e6ad1873df448eef/gemma/config.py#L118).

Reply:
yep the PR is here #31775 we'll do a patch for it